packages/plugin-nm/sources/NodeModulesLinker.ts

```diff
@@ -664,21 +670,24 @@ const buildLocationTree = (locatorMap: NodeModulesLocatorMap | null, {skipPrefix
   return locationTree;
 };
 
-const symlinkPromise = async (srcPath: PortablePath, dstPath: PortablePath) => {
-  let stats;
-
-  try {
-    if (process.platform === `win32`) {
+const symlinkPromise = async (srcPath: PortablePath, dstPath: PortablePath, nmFolderLinkMode: NodeModulesFolderLinkMode) => {
+  // use junctions on windows if in classic mode
+  if (process.platform === `win32` && nmFolderLinkMode === NodeModulesFolderLinkMode.CLASSIC) {
+    let stats;
+    try {
       stats = await xfs.lstatPromise(srcPath);
+    } catch (e) {
     }
-  } catch (e) {
-  }
 
-  if (process.platform == `win32` && (!stats || stats.isDirectory())) {
-    await xfs.symlinkPromise(srcPath, dstPath, `junction`);
-  } else {
-    await xfs.symlinkPromise(ppath.relative(ppath.dirname(dstPath), srcPath), dstPath);
+    if (process.platform == `win32` && (!stats || stats.isDirectory())) {
```

Comment:
```suggestion
    if (!stats || stats.isDirectory()) {
```
The "parent" if statement already checked that it's `win32`

Reply:
Thank you! Fixed!